ReSharper.FSharp/src/FSharp.Psi.Features/src/QuickFixes/AddExtensionAttributeFix.fs

```diff
@@ -0,0 +1,51 @@
+﻿namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Util
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Util.FSharpAttributesUtil
+
+type AddExtensionAttributeFix(warning: ExtensionMemberInNonExtensionTypeWarning) =
+    inherit FSharpQuickFixBase()
+
+    let [<Literal>] extensionNamespaceName = "System.Runtime.CompilerServices"
+    let [<Literal>] attributeName = "Extension"
+    let declaration =
+        let attributeHolder: IFSharpTypeMemberDeclaration =
+            match LetModuleDeclNavigator.GetByAttribute(warning.Attr) with
+            | null -> MemberDeclarationNavigator.GetByAttribute(warning.Attr) :> _
+            | letModuleDec -> letModuleDec :> _
+        attributeHolder.GetContainingTypeDeclaration()
+
+    override x.Text = sprintf "Add 'Extension' attribute to '%s'" declaration.SourceName
+
+    override x.IsAvailable _ =
+        match declaration with
+        | :? ITopLevelModuleLikeDeclaration -> false
+        | null -> false
```

Comment:
Is this check really needed?

Reply:
yes, since MemberDeclarationNavigator.GetByAttribute can return null

Reply:
But `isValid` does the check too. :)

Reply:
hmm, really